instrumentation/que/lib/opentelemetry/instrumentation/que/patches/que_job.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Que
+      module Patches
+        # Module to prepend to Que::Job for instrumentation
+        module QueJob
+          def self.prepended(base)
+            class << base
+              prepend ClassMethods
+            end
+          end
+
+          # Module to prepend to Que singleton class
+          module ClassMethods
+            def enqueue(*args, tags: nil, **arg_opts)
+              tracer = Que::Instrumentation.instance.tracer
+              otel_config = Que::Instrumentation.instance.config
+
+              tracer.in_span('send', kind: :producer) do |span|
+                # Que doesn't have a good place to store metadata. There are
+                # basically two options: the job payload and the job tags.
+                #
+                # Using the job payload is very brittle. We'd have to modify
+                # existing Hash arguments or add a new argument when there are
+                # no arguments we can modify. If the server side is not using
+                # this instrumentation yet (e.g. old jobs before the
+                # instrumentation was added or when instrumentation is being
+                # added to client side first) then the server can error out due
+                # to unexpected payload.
+                #
+                # The second option (which we are using here) is to use tags.
+                # They also are not meant for tracing information but they are
+                # much safer to use than modifying the payload.
+                if otel_config[:propagation_style] != :none
+                  tracing_info = {}
+                  OpenTelemetry.propagation.inject(tracing_info)
+
+                  tags ||= []
+                  tags += tracing_info.map { |k, v| "#{k}:#{v}" }
```

Comment:
You can define your own `setter` that appends to `tags` directly, rather than create a `Hash` and map it. See https://github.com/open-telemetry/opentelemetry-ruby/blob/7e314bde62bb0fb46357b459e726ff93b8737795/api/lib/opentelemetry/context/propagation/text_map_setter.rb#L13-L17
```suggestion
                  tags ||= []
                  OpenTelemetry.propagation.inject(tags, setter: tag_setter)
```
where:
```ruby
class TagSetter
  def set(carrier, key, value)
    carrier << "#{key}:#{value}"
  end
end

tag_setter = TagSetter.new
```

Reply:
https://github.com/open-telemetry/opentelemetry-ruby/compare/21b528af11288591298b67f656aacb29bf90444f..b828d9146b67d623f84b6b963c7b8bd8abb302ee